src/SDKs/Azure.Base/data-plane/Azure.Base/Http/Pipeline/HttpClientTransport.cs

```diff
@@ -27,21 +27,37 @@ public HttpClientTransport(HttpClient client = null)
         public sealed override HttpPipelineRequest CreateRequest(IServiceProvider services)
             => new PipelineRequest();
 
-        public sealed override async Task ProcessAsync(HttpPipelineMessage message)
+        public override async Task ProcessAsync(HttpPipelineMessage message)
         {
-            var pipelineRequest = message.Request as PipelineRequest;
-            if (pipelineRequest == null) throw new InvalidOperationException("the request is not compatible with the transport");
+            using (HttpRequestMessage httpRequest = BuildRequestMessage(message))
+            {
+                SetResponse(message, await _client.SendAsync(httpRequest, message.Cancellation).ConfigureAwait(false));
+            }
+        }
 
-            using (HttpRequestMessage httpRequest = pipelineRequest.BuildRequestMessage(message.Cancellation))
+        public override void Process(HttpPipelineMessage message)
+        {
+            using (HttpRequestMessage httpRequest = BuildRequestMessage(message))
             {
-                HttpResponseMessage responseMessage = await ProcessCoreAsync(message.Cancellation, httpRequest).ConfigureAwait(false);
-                message.Response = new PipelineResponse(message.Request.RequestId, responseMessage);
+                var client = new SyncHttpClientHandler();
```

Comment:
Can we remove this type and just do sync over async? In the test project we should write a full blown transport using just the socket handler

Reply:
It does sync over async, `HttpClientHandler.SendAsync` is not public so I had to add a method to pass through.

Reply:
> In the test project we should write a full blown transport using just the socket handler

Why do it in the test project?